pytorch_lightning/trainer/trainer.py

```diff
@@ -122,7 +122,7 @@ def __init__(
         accumulate_grad_batches: Union[int, Dict[int, int]] = 1,
         max_epochs: Optional[int] = None,
         min_epochs: Optional[int] = None,
-        max_steps: Optional[int] = None,
+        max_steps: Optional[int] = -1,
```

Comment:
I think it's better for us to remove the `Optional` now, so it's easier for users to know they should change the value

```suggestion
        max_steps: int = -1,
```